releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java

```diff
@@ -19,65 +19,136 @@
 
 package com.github.checkstyle.templates;
 
+import static com.github.checkstyle.globals.Constants.BREAKING_COMPATIBILITY_LABEL;
+import static com.github.checkstyle.globals.Constants.BUG_LABEL;
+import static com.github.checkstyle.globals.Constants.MISCELLANEOUS_LABEL;
+import static com.github.checkstyle.globals.Constants.NEW_FEATURE_LABEL;
+import static com.github.checkstyle.globals.Constants.NEW_MODULE_LABEL;
+import static org.junit.Assert.fail;
+import static org.kohsuke.github.GHIssueState.CLOSED;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.IOException;
-import java.lang.reflect.Field;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 import java.util.Arrays;
-import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Locale;
+import java.util.Set;
 
 import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.kohsuke.github.GHFileNotFoundException;
 import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHRepository;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
-import com.github.checkstyle.CliOptions;
-import com.github.checkstyle.CliOptions.Builder;
+import com.github.checkstyle.Main;
 import com.github.checkstyle.MainProcess;
-import com.github.checkstyle.globals.Constants;
-import com.github.checkstyle.globals.ReleaseNotesMessage;
-import com.google.common.collect.ArrayListMultimap;
-import com.google.common.collect.Multimap;
+import com.github.checkstyle.git.CsGit;
+import com.github.checkstyle.github.CsGitHub;
+import com.github.checkstyle.internal.GhIssueUtil;
+import com.github.checkstyle.internal.RevCommitUtil;
+import com.github.checkstyle.publishers.TwitterPublisher;
 
 public class TemplateProcessorTest {
+    private static final Set<RevCommit> TEST_COMMITS = new LinkedHashSet<>();
```

Comment:
Had to use a `LinkedHashSet` instead of just `HashSet` since order of commits(and output in turn) was randomized on every run making it impossible to assert.

Reply:
commits is List - just a sequence of commits of strict order.
it is not a set.

Reply:
@romani 
The reason this is a set is because it is what is directly returned from the production code at
https://github.com/checkstyle/contribution/blob/a0e6b5abf8c9466e5eae021cd1d6fd7902b834a4/releasenotes-builder/src/main/java/com/github/checkstyle/git/CsGit.java#L57

Even that method returns a LinkedHashSet so another issue would have to be created for it to be turned to a list if that is what is wanted.